cbackup.sh

```diff
@@ -277,7 +277,7 @@ function do_backup() {
         fi
 
         # Battery optimization
-        if grep -q "$app" /data/system/deviceidle.xml; then
+        if [[ -f /data/system/deviceidle.xml ]] && grep -q "$app" /data/system/deviceidle.xml; then
```

Comment:
This could just be `grep -q [...] 2> /dev/null`, so there's no need for an extra check.